ip_reader.c

```diff
@@ -0,0 +1,200 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include "ip_reader.h"
+
+#define OCTET_1ST(ip) ((ip) >> 24)
+#define OCTET_2ND(ip) OCTET_1ST((ip) << 8)
+#define OCTET_3RD(ip) OCTET_1ST((ip) << 16)
+#define OCTET_4TH(ip) OCTET_1ST((ip) << 24)
+#define DECIMAL_FORMAT_STRING "%u.%u.%u.%u"
+#define DECIMAL_FORMAT_VALUES(ip) OCTET_1ST(ip),OCTET_2ND(ip),OCTET_3RD(ip),OCTET_4TH(ip)
+
+enum return_flags {SUCCESS, ADDR_INVAL, MASK_INVAL, FORM_INVAL, EXIT_FLAG};
+
+int parse_ip4_short(struct ip4_address *address, char *input)
+{
+	if (read_data(&address->addr, input)) {
+		printf("Invalid IP address!\n");
+		return ADDR_INVAL;
+	}
+
+	if (demask(address, input)) {
+		printf("Not a valid subnet mask!\n");
+		return MASK_INVAL;
+	}
+	return SUCCESS;
+}
+
+int parse_ip4_long(struct ip4_address *address, char *input[])
+{
+	if (read_data(&address->addr, input[1])) {
+		printf("Invalid IP address!\n");
+		return ADDR_INVAL;
+	}
+
+	if (read_data(&address->mask, input[2])) {
+		printf("Not a valid subnet mask!\n");
+		return MASK_INVAL;
+	}
+
+	return SUCCESS;
+}
+
+int read_data(uint32_t *temp_name, char *input)
+{
+	if (format_valid(input)) return FORM_INVAL;
```

Comment:
"if input's format is valid return FORM_INVAL"
With function names and behavior we need to be careful. Code should be similar to sentences, easy to read and verbose about its actions. With "valid" in function name I would expect from function to return true/false where true is a correct value. If for some reasons you really need to return status I'd suggest naming function check_format, I think that's more common convention.

Reply:
Fixed.